content/doc/book/glossary/index.adoc

```diff
@@ -5,15 +5,119 @@ layout: chapter
 :author:
 :email: jenkinsci-users@googlegroups.com
 :sectanchors:
-:toc: left
-:notitle:
+:toc:
+
+////
+XXX: Pages to mark as deprecated by this document:
+      * https://wiki.jenkins-ci.org/display/JENKINS/Terminology
+////
 
 [glossary]
 = Glossary
 
+////
+NOTE: The [glossary] delimiter in AsciiDoctor doesn't autogenerate anchors for
+each of the terms below. Which means that if we want to cross-reference terms
+directly from other documents we need to include an inline anchor.
+
+Additionally, because these inline anchors don't attach to section headings,
+cross referencing must include the appropriate display text, for example:
+
+  MyTerm:: [[myterm]] this is the definition of MyTerm
+
+Should be cross-referenced with:
+
+  <<myterm,MyTerm>>
+
+To ensure it is rendered appropriately.
+////
+
+== General Terms
 
 [glossary]
+Agent::  [[agent]]
+    An agent is typically a machine, or container, which connects to a Jenkins
+    master and executes tasks when directed by the master.
+Artifact:: [[artifact]]
+    An immutable file generated during a <<build,Build>> or <<pipeline,Pipeline>>
+    run. Artifacts are typically *archived* onto the Jenkins <<master,Master>> for
+    later retrieval by users.
```

Comment:
uploading a file into Artifactory doesn't relate to the concept of an Artifact in Jenkins' vocabulary though. I think this would be clearer defining artifact only as a file which is archived to the master, otherwise it's just any ol' file.